library/VTKExtensions/Readers/vtkF3DMemoryMesh.cxx

```diff
@@ -0,0 +1,74 @@
+#include "vtkF3DMemoryMesh.h"
+
+#include "vtkDataArrayRange.h"
+#include "vtkFloatArray.h"
+#include "vtkInformationVector.h"
+#include "vtkObjectFactory.h"
+#include "vtkPointData.h"
+#include "vtkSMPTools.h"
+
+#include <numeric>
+
+vtkStandardNewMacro(vtkF3DMemoryMesh);
+
+//------------------------------------------------------------------------------
+vtkF3DMemoryMesh::vtkF3DMemoryMesh()
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//------------------------------------------------------------------------------
+vtkF3DMemoryMesh::~vtkF3DMemoryMesh() = default;
+
+//------------------------------------------------------------------------------
+void vtkF3DMemoryMesh::SetPoints(const std::vector<float>& positions)
+{
+  vtkNew<vtkFloatArray> arr;
+  arr->SetNumberOfComponents(3);
+  arr->SetNumberOfTuples(positions.size() / 3);
+
+  std::copy(positions.cbegin(), positions.cend(), arr->GetPointer(0));
+
+  vtkNew<vtkPoints> points;
+  points->SetDataTypeToFloat();
+  points->SetData(arr);
+
+  this->Mesh->SetPoints(points);
+}
+
+//------------------------------------------------------------------------------
+void vtkF3DMemoryMesh::SetCells(
+  const std::vector<unsigned int>& cellSize, const std::vector<unsigned int>& cellIndices)
+{
+  vtkNew<vtkIdTypeArray> offsets;
+  vtkNew<vtkIdTypeArray> connectivity;
+  offsets->SetNumberOfTuples(cellSize.size() + 1);
+  connectivity->SetNumberOfTuples(cellIndices.size());
+
+  // fill offsets
+  offsets->SetTypedComponent(0, 0, 0);
+  std::inclusive_scan(cellSize.cbegin(), cellSize.cend(),
+    vtk::DataArrayValueRange(offsets).begin() + 1,
+    [](unsigned int a, unsigned int b) { return static_cast<vtkIdType>(a + b); });
+
```

Comment:
not covered by tests ?

Reply:
That's because there's a single cell. I've fixed it.